### PR TITLE
Musark 467

### PR DESCRIPTION
**Pull request by rituvesh**

### CODE REVIEW

File: src/reducers/grid/observationcontrol/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import Config from '../../../config'
 import { apiUrl } from '../../../util'
+import { sortObject } from '../../../util/sort'
```
Comment (jarlah) [inline src/reducers/grid/observationcontrol/index.js]: do this for all containers. remove sort from reducers and place it in containers.

this is example magasin grid container.

```
import { createSelector } from 'reselect'

import { sortObject } from '../../../util/sort'

const getStorageUnits = (state) => state.storageGridUnit.data || []

const getSortedStorageUnits = createSelector(
  [ getStorageUnits ],
  (storageUnits) => sortObject(sortObject(storageUnits, 'name'), 'type')
)

const mapStateToProps = (state) => ({
  ....
  children: getSortedStorageUnits(state),
  ....
});
```

Comment (jarlah) [reply src/reducers/grid/observationcontrol/index.js]: https://github.com/reactjs/reselect

**Review by jarlah — CHANGES_REQUESTED** [review]
Move sorting from reducers to containers

File: src/reducers/storageobject/grid/__tests__/StorageObjectReducer-test.js [hunk 2]
```diff
@@ -34,16 +31,7 @@ const comingFromBackend = [
   }
 ]
 
-
 describe('StorageUnitReducer', () => {
-  it('Initial state is set', () => {
```
Comment (jarlah) [inline src/reducers/storageobject/grid/__tests__/StorageObjectReducer-test.js:39]: why have you removed this test

File: src/util/sort.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+export const sortObject = (obj: any, key: string, inputKeyType: string = '', sortAscending: bool = true, keyChild: string): any => {
```
Comment (jarlah) [inline src/util/sort.js]: why keep this method? remove it if not used (it should not be any need for this)

**Review by jarlah — CHANGES_REQUESTED** [review]

File: src/reducers/storageobject/grid/__tests__/StorageObjectReducer-test.js (hunk 2 above)
Comment (rituvesh) [reply src/reducers/storageobject/grid/__tests__/StorageObjectReducer-test.js:39]: I have created new one for LOAD_SEVERAL_SUCCESS. This was the old one. But if you want I can add it back.

Comment (jarlah) [reply src/reducers/storageobject/grid/__tests__/StorageObjectReducer-test.js:39]: no its fine. I just need the reason :) thx .. then its 👍

**Review by jarlah — APPROVED** [review]

**Pull request merged**